### PR TITLE
feat(musa): add MUSA device support

### PR DESCRIPTION
**Pull request by wenqf11**
## Motivation

Add MUSA device(https://github.com/MooreThreads) support

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.io/docs/developer_guide/contribution_guide).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.io/docs/developer_guide/contribution_guide).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.io/docs/developer_guide/contribution_guide).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.io/docs/developer_guide/benchmark_and_profiling) and [Accuracy Results](https://docs.sglang.io/docs/developer_guide/evaluating_new_models).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

**Comment by jiapingW** [comment]
@wenqf11  Can you add your training results that train used by MUSA device?

**Comment by wenqf11** [comment]
@jiapingW 

training configuration:
```model:
  target_model_path: models/Qwen3.6-35B-A3B 
  draft_checkpoint_path: outputs/qwen3.6-35b-a3b-dflash2-disaggregated-failed/qwen3.6-35b-a3b-dflash2-disaggregated-latest/
  draft_model_config: configs/qwen3.6-35b-a3b-dflash2.json
  target_backend: sglang
  embedding_key: model.language_model.embed_tokens.weight
  torch_dtype: bfloat16
  sglang_mem_fraction_static: 0.9

data:
  train_data_path: datasets/eaglechat700k_chinese_qwen3.6-35b-a3b.jsonl 
  max_length: 4096
  chat_template: qwen3.5
  build_dataset_num_proc: 64
  cache_dir: cache
  cache_key: eaglechat700k_chinese

training:
  strategy: dflash
  num_epochs: 10
  max_steps: 100000
  batch_size: 2
  learning_rate: 0.0006 
  warmup_ratio: 0.04
  max_grad_norm: 1
  attention_backend: sdpa
  num_anchors: 512
  loss_decay_gamma: 7
  lk_loss_type: lambda
  loss_type: dpace
  objective_chunk_blocks: 128
  dflash2_selector_loss_alpha: 1.0
  dflash2_selector_warmup_ratio: 0.0005
  dflash2_selector_ramp_ratio: 0.0005
  save_interval: 1000
  log_interval: 50
  dist_timeout: 30
  seed: 42

tracking:
  report_to: swanlab 
  swanlab_project: "qwen36-35b-a3b-dflash2-disagg"
  swanlab_name: "qwen36-35b-a3b-dflash2-server-captured"
  swanlab_key: "c9KwlgA5dcwZIBhsx7OY3"

run_id: qwen3.6-35b-a3b-dflash2-disaggregated
output_dir: outputs/qwen3.6-35b-a3b-dflash2-disaggregated

deployment:
  mode: disaggregated
  trainer:
    nnodes: 1
    nproc_per_node: 4
  disaggregated:
    control_dir: outputs/qwen3.6-35b-a3b-dflash2-disaggregated/control
    consumer_state_dir: outputs/qwen3.6-35b-a3b-dflash2-disaggregated/consumer-state
    backend: mooncake
    server_urls:
      - http://127.0.0.1:30000
    mooncake_metadata_server: http://127.0.0.1:35880/metadata
    mooncake_master_server_addr: 127.0.0.1:35551
    mooncake_protocol: tcp
```

training loss and acc:
image

benchmark **gsm8k_zh** with final checkpoints:
```=== backend=flashinfer tp=2 (DFLASH) ===
command=sglang serve --model-path /jfs/models/Qwen/Qwen3.6-35B-A3B --trust-remote-code --attention-backend flashinfer --tp-size 2 --dtype bfloat16 --mem-fraction-static 0.9 --max-running-requests 1 --cuda-graph-bs 1 --cuda-graph-max-bs 1 --speculative-algorithm DFLASH --speculative-dflash-block-size 16 --speculative-draft-model-path /jfs/qingfu.wen/models/qwen3.6-35b-a3b-dflash2-step100000 --disable-radix-cache --device cuda --host 127.0.0.1 --port 20785
CI_OFFLINE: Launching server HF_HUB_OFFLINE=0 model=/jfs/models/Qwen/Qwen3.6-35B-A3B
[DFLASH]   conc= 1 n=32   toks/s=401.88 latency=114.5s acc=0.781 invalid=0.000 accept_len=6.668 spec_verify_ct_sum=7590
```

**Comment by wenqf11** [comment]
@jiapingW  pls take a look

**Comment by FrankLeeeee** [comment]
Hi @wenqf11 , besides the distributed communication backend, can you also provide the dependencies for musa so that we can add them to #861 . Moreoever, how should we set up the CI for MUSA?

**Comment by wenqf11** [comment]
@FrankLeeeee  For SpecForge on MUSA, it depends on SGLang and Mooncake, both of which have supported MUSA devices.
Install SGLang: https://docs.sglang.io/docs/sglang-diffusion/installation#platform-specific-musa-moore-threads-gpus
Install Mooncake: pip install mooncake-transfer-engine-musa
Install Specforge:
```
git clone https://github.com/sgl-project/SpecForge.git
cd SpecForge
python -m pip install -e . --no-deps
```

As for CI, I am not sure. Can we refer to CI for ROCm and Ascend?
